commands/run.go

```diff
@@ -0,0 +1,207 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"os/exec"
+
+	"github.com/openfaas/faas-cli/stack"
+	"github.com/spf13/cobra"
+)
+
+const localSecretsDir = ".secrets"
+
+func init() {
+	faasCmd.AddCommand(newRunCmd())
+}
+
+type runOptions struct {
+	print    bool
+	port     int
+	network  string
+	extraEnv map[string]string
+	output   io.Writer
+	err      io.Writer
+}
+
+func newRunCmd() *cobra.Command {
+	opts := runOptions{}
+
+	cmd := &cobra.Command{
+		Use:   `run NAME -f YAML_FILE`,
+		Short: "User docker to run a function locally, this can be used for quick local testing.",
+		Example: `
+		faas-cli run stronghash
```

Comment:
In my opinion, encouraging users to create secrets in the function source code folder may lead to committing these to source control. I know that I've done it in the past.

If someone runs `faas-cli run` and we're also directing them to create a .secrets folder.

Thoughts on updating the .gitignore file?